### DISCUSSION TITLE
Décalage de 2 degrès entre Preset dans HA et la consigne réelle de l'app Cozytouch

### DISCUSSION THREAD

**Discussion by Yonizzer** [jmcollin78/versatile_thermostat#312] [Q&A] [answered]
Bonjour,
Je viens de découvrir ton plugin, et c'est vraiment top👍

J'ai configuré plusieurs thermostats 'climate' , et je viens de me rendre compte qu'entre les consignes du preset (paramétré dans l'intégration HA) et celle réellement mise, il y avait un décalage de 2 degrès.

Saurais-tu d'où cela peut provenir ?

Information de l'installation:
- Radiateur Sauter, connecté via Cozytouch, la programmation est active (éco et confort paramétré également) via les heures.
- je les ai ajouté dans HA, (juste pour 'regarder' pour l'instant)
Merci à toi

**Comment by jmcollin78** [comment]
Hello,

Je suppose qu'il s'agit de ça : https://github.com/jmcollin78/versatile_thermostat/blob/main/README-fr.md#lauto-r%C3%A9gulation

**Reply by Yonizzer** [reply]
Effectivement j'ai regardé cette partie, mais je n'ai pas l'impression que cela réponde totalement à mon besoin. Ou alors je n'ai pas tout saisi!
Si je peux ajouter un peu de précision:

Radiateur Salon (2 radiateurs)
- Température sonde 1 : 16.10 °C (valeur affichée dans HA), 17 °C affiché dans l'app CozyTouch.
- Température sonde 2 : 16.50 °C (valeur affichée dans HA), 17.5 °C affiché dans l'app CozyTouch.

la Consigne Eco : 17 °C
--> radiateur est bien actif, comme la sonde est proche du mur et inférieur a 17 °C.

Le Thermomètre Salon - Température sonde : 19 °C (température de la pièce)
--> il ne devrait pas chauffer (du moins j'aimerai ^^)

J'ai configuré le plugin en mettant comme référence le "Thermomètre Salon"
En mettant l'option Auto-Régulation a "Moyenne", la consigne oscille entre +1.5°C à +2°C. donc c'est good pour cela.

**Reply by jmcollin78** [reply]
Du coup, y a un soucis ou pas ? J'arrive pas à savoir.
Si le thermomètre est à 19° (celui que tu as configuré dans le VTherm je suppose) et que la consigne est à 17°, ca ne devrait pas chauffer du tout.

Il ne faut pas mettre les températures des sondes des radiateurs dans le VTherm mais bien les températures du thermomètre externe aux radiateurs (thermomètre salon si j'ai bien compris).

Si j'ai rien compris (toujours possible) essaye de me donner les attributs de ton VTherm (dans outils de dev / etat)

**Reply by Yonizzer** [reply]
Dans le VTherm j'ai bien mis le thermomètre externe au radiateur, ça on est bon.

Voici le VTherm
![image](https://github.com/jmcollin78/versatile_thermostat/assets/15137844/dc508d7e-1ade-41db-90a4-ee8d87f745b7)
 T° Actuel: 19.5
 Consigne: 16

Et pourtant les 2 radia sous-jacent a ce VTherm, chauffent plus ou moins quand même car les T° qu'ils ont réellement sont respectivement:
![image](https://github.com/jmcollin78/versatile_thermostat/assets/15137844/a9ca005b-a3b0-4638-b78f-5dd408fb9eaa)

D'ou mon étonnement, mais peut être ai-je mal interprété les fonctionnalité du plugin, et je me trompe, c'est clairement possible!
Mes radiateur sont déjà "connecté", peut-etre que la programmation de CozyTouch prend le pas dessus.. j'ai du mal à savoir.
Je vais peut-être supprimer mes 2 radiateurs (climate.salon, climate.sam)

**Reply by jmcollin78** [reply]
Tu veux qu'il fait 19° dans la pièce mais que le radiateur mesure 15,7° ?
Si c'est bien ce que je comprends, comme c'est le radiateur qui régule en mode over_climate, on ne va pas pouvoir faire grand chose. Comment c'est possible d'avoir plus de 3° d'écart ?

EDIT:
Si,  le truc qui va te sauver c'est d'activer l'auto-régulation en mode Forte. La température de consigne sur tes radiateurs va baisser de beaucoup (jusqu'à 5°) tant que la temp de la pièce est au dessus de la consigne. Ca, ca va marcher.

**Reply by Yonizzer** [reply]
J’ai un poêle qui chauffe le salon, donc le thermomètre du salon à la bonne température en quelque sorte, pour le ressenti.
Les 2 radiateurs étant collés au mur sont plus bas en T*. Du coup même quand il faut 19° dans la pièce, les sondes des radiateurs n’augmentent pas tant que ça, surtout quand il fait froid dehors.. et l’isolation de la maison doit y faire aussi ! Enfin bref.

Il y a déjà une marge appliquée dans l’app de Sauter, il arrondi à 1° supérieur. Mais ça je ne le savais pas avant d’ajouter les radiateur dans HA. La température dans l’app affiche 16°, la valeur réelle dans HA est de 15.1°.

J'aimerai m’affranchir de leur app vraiment pas opti, et les gérer grâce à ce Thermomètre Externe du salon..

Je vais regarder pour l’auto-régulation en mode « Fort »

**Reply by jmcollin78** [reply, marked as answer]
Oui l'auto-regulation est la solution. Regarde ici, même soucis et même solution: https://github.com/jmcollin78/versatile_thermostat/issues/316

**Reply by Yonizzer** [reply]
Merci pour ton aide 👍

**Comment by jmcollin78** [comment]
https://github.com/jmcollin78/versatile_thermostat/blob/main/README-fr.md#le-radiateur-chauffe-alors-que-la-temp%C3%A9rature-de-consigne-est-d%C3%A9pass%C3%A9e-ou-ne-chauffe-pas-alors-que-la-temp%C3%A9rature-de-la-pi%C3%A8ce-est-bien-en-dessous-de-la-consigne
